### DISCUSSION TITLE
Podman extremely slow on raspberry pi after OS upgrade

### DISCUSSION THREAD

**Discussion by endlisnis** [containers/podman#21111] [Q&A] [answered]
I was running podman in Debian 11 for many months and got decent performance out of it.  It would take a few seconds to start a container; and a few minutes to build a new image.

When I upgraded to Debian 12, podman became about 50x slower.  Just starting a container would take about 3 minutes and building a 30-step image would take hours.

I even tried a full OS reinstall using Debian 12 to avoid upgrade issues, and I saw the same result.

I'm running podman version 4.3.1 currently.  Before the upgrade, I was running v3.something.

I tried running the same build script and starting the same containers using docker on the same raspberry pi; and it's performance is similar to podman in Debian 11.

Any ideas what's going on?

I'm running a raspberry pi 400 (4G) on an up-to-date Debian 12.

```
rolf@raspberrypi ~
$ time sudo docker run -it --rm fedora /bin/echo


real    0m1.341s
user    0m0.005s
sys     0m0.014s

rolf@raspberrypi ~
$ time podman run -it --rm fedora /bin/echo


real    0m17.962s
user    0m0.744s
sys     0m3.423s
```
(I ran these several times to get all the caches populated. I only posted the last runs).

Even just "exec" into an existing container takes dozens of seconds.

For extra details, after my OS upgrade, I could start and exec into an image/container with no obvious slowdown. It was only newly created images that suffered from this slowdown.

**Comment by vrothberg** [comment]
Can you share the output of `podman info`?

**Reply by endlisnis** [reply]
[podmaninfo.txt](https://github.com/containers/podman/files/13811652/podmaninfo.txt)

**Comment by rhatdan** [comment]
When I hear slowness, I usually think vfs storage driver versus overlay.

**Reply by endlisnis** [reply]
Do you mean "graphDriverName: vfs"?  Which one is faster/better?

**Reply by rhatdan** [reply]
Overlay is much faster.

**Reply by rhatdan** [reply, marked as answer]
Edit storage.conf and do a `podman system reset`

**Reply by rhatdan** [reply]
Also make sure the arch for the Rasberry PI and the images you are running testing match.  Don't run x86 images on an arm arch.

**Reply by endlisnis** [reply]
If overlay is much faster, why is vfs the default?

**Reply by endlisnis** [reply]
Changing it to "overlay" made it even faster than docker:
```
$ time podman run -it --rm fedora /bin/echo


real    0m0.648s
user    0m0.163s
sys     0m0.105s
```

**Reply by rhatdan** [reply]
What OS are you installing?  On Fedora the default is overlay.

VFS is supported everwhere, overlay not so much, but we set it in storage.conf for distributions that we control.

**Reply by endlisnis** [reply]
I'm running Raspberry Pi OS 5 (Debian 12).

I was running Raspberry Pi OS 4 (Debian 11) before, and it didn't have this problem.  Maybe the default there was overlay?

**Reply by rhatdan** [reply]
Don't know, Sadly we seem to go with least common denominator, we could look into trying overlay first and failing over to VFS if the user did not specify a default.  Would  you open an issue on containers/storage?

**Reply by endlisnis** [reply]
I opened it here: https://github.com/containers/storage/issues/1805
